lib/middlewares/authenticated_request_test.go

```diff
@@ -0,0 +1,80 @@
+package middlewares_test
+
+import (
+	"net/http"
+
+	"github.com/markgravity/golang-ic/lib/api/v1/controllers"
+	"github.com/markgravity/golang-ic/lib/middlewares"
+	"github.com/markgravity/golang-ic/test"
+	"github.com/markgravity/golang-ic/test/fabricators"
+	"github.com/markgravity/golang-ic/test/helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AuthenticatedRequest", func() {
+	AfterEach(func() {
+		test.CleanUpDatabase()
+	})
+
+	Describe("HandleAuthenticatedRequest", func() {
+		Context("given VALID headers", func() {
+			It("returns status OK", func() {
+				user := fabricators.FabricateUser("test@gmail.com", "123456")
+				accessToken := "Bearer " + helpers.GenerateToken(user.Base.ID.String())
+				headers := map[string]string{
+					"Authorization": accessToken,
+				}
+
+				ctx, response := test.MakeRequest(http.MethodGet, "/", headers, nil)
+
+				middlewares.HandleAuthenticatedRequest()(ctx)
+
+				Expect(response.Result().StatusCode).To(Equal(http.StatusOK))
+			})
+
+			It("sets data to context", func() {
```

Comment:
Should we be more precise about what data? This test description does not help understand the code. 

```suggestion
			It("sets the 'user' key with the current user id, into the context", func() {
```